CMakeLists.txt

```diff
@@ -20,30 +21,53 @@ include(${_project_options_SOURCE_DIR}/Index.cmake)
 # Set the project name and language
 project(myproject LANGUAGES CXX)
 
+# Set up some options for analysis which is desirable to default to on, but
+# also desirable to disable for some types of builds
+option(ENABLE_CPPCHECK "Enable cppcheck static analysis during compilation" ON)
+option(ENABLE_CLANG_TIDY "Enable clang-tidy static analysis during compilation" ON)
+option(ENABLE_ADDRESS_SANITIZER "Enable address-sanitizer on compilers that support it" ON)
+option(ENABLE_UNDEFINED_BEHAVIOR_SANITIZER "Enable undefined-behavior sanitizers on compilers that support it" ON)
+
+if(ENABLE_CPPCHECK)
+  set(CPPCHECK "ENABLE_CPPCHECK")
+endif()
+
+if(ENABLE_CLANG_TIDY)
+  set(CLANG_TIDY "ENABLE_CLANG_TIDY")
+endif()
+
+if(ENABLE_ADDRESS_SANITIZER)
+  set(ADDRESS_SANITIZER "ENABLE_SANITIZER_ADDRESS")
+endif()
+
+if(ENABLE_UNDEFINED_BEHAVIOR_SANITIZER)
+  set(UNDEFINED_BEHAVIOR_SANITIZER "ENABLE_SANITIZER_UNDEFINED_BEHAVIOR")
+endif()
+
```

Comment:
```suggestion
 # Add global CMake options
include(${_project_options_SOURCE_DIR}/src/GlobalOptions.cmake)
```

Reply:
I've I'm reading your changes correctly, it would default to all of the analysis being off, is that correct?

Also, Windows supports address sanitizer now.

Reply:
The default can be changed here:
https://github.com/cpp-best-practices/project_options/blob/6d1758261af3e7c486522760ebd3a8fb0fcbecb2/src/GlobalOptions.cmake#L21

Reply:
Right, so the options are: *everything* is default `ON`, or *everything* is default `OFF` is that correct?

Reply:
> Also, Windows supports address sanitizer now.

Yes, it does. But it requires running vcvarsall (opening the VS command prompt) so that the sanitizer libraries are on the path during runtime.

Reply:
> Right, so the options are: everything is default ON, or everything is default OFF is that correct?

Yes. But the global options are not used by default. Only they are if the user added `${}` around the arguments. This allows granular dynamic options (to keep the core options declarative).

https://github.com/cpp-best-practices/project_options#changing-the-project_options-parameters-dynamically

```
project_options(
	 # ...
      # ❗ Now, the address and undefined behavior sanitizers are enabled through CMake options
      ${ENABLE_SANITIZER_ADDRESS}
      ${ENABLE_SANITIZER_UNDEFINED_BEHAVIOR}
      # ENABLE_SANITIZER_LEAK
      # ENABLE_SANITIZER_THREAD
      # ENABLE_SANITIZER_MEMORY

)
```

Reply:
> It actually makes sense to change the default. Because the user would opt-in to use global options by adding ${} around the arguments.

Interesting point

Reply:
My argument here for choosing default OFF was that the core options are already declared through function arguments, and the default build options are for production. A library author does not want to enable sanitizers during the release.

Reply:
Oh, also there is no sanitizer support on Windows with Clang. The MSVC support is there but as I said requires running code in vcvarsall enabled shell.